packages/nextjs/src/config/loaders/dataFetchersLoader.ts

```diff
@@ -94,44 +105,73 @@ function wrapFunctions(userCode: string, templateCode: string, filepath: string)
-  // We know one or the other will be defined, depending on the version of webpack being used
-  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-  const { projectDir } = this.getOptions ? this.getOptions() : this.query!;
-
   // For now this loader only works for ESM code
   if (!isESM(userCode)) {
     return userCode;
   }
 
-  // If none of the functions we want to wrap appears in the page's code, there's nothing to do. (Note: We do this as a
-  // simple substring match (rather than waiting until we've parsed the code) because it's meant to be an
-  // as-fast-as-possible fail-fast. It's possible for user code to pass this check, even if it contains none of the
-  // functions in question, just by virtue of the correct string having been found, be it in a comment, as part of a
-  // longer variable name, etc. That said, when we actually do the code manipulation we'll be working on the code's AST,
-  // meaning we'll be able to differentiate between code we actually want to change and any false positives which might
-  // come up here.)
-  if (Object.keys(DATA_FETCHING_FUNCTIONS).every(functionName => !userCode.includes(functionName))) {
-    return userCode;
-  }
+  // We know one or the other will be defined, depending on the version of webpack being used
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const { projectDir } = 'getOptions' in this ? this.getOptions() : this.query;
 
-  const templatePath = path.resolve(__dirname, '../templates/dataFetchersLoaderTemplate.js');
-  // make sure the template is included when runing `webpack watch`
-  this.addDependency(templatePath);
+  // Proxy the processed file
+  if (!this.resourceQuery.includes('sentry-proxy-loader')) {
+    const ast = makeAST(userCode, true); // is there a reason to ever parse without typescript?
 
-  const templateCode = fs.readFileSync(templatePath).toString();
+    const exportedIdentifiers = getExportIdentifiers(ast);
 
-  const [modifiedUserCode, modifiedTemplateCode] = wrapFunctions(
-    userCode,
-    templateCode,
-    // Relative path to the page we're currently processing, for use in error messages
-    path.relative(projectDir, this.resourcePath),
-  );
+    let outputFileContent = '';
 
-  // Fill in template placeholders
-  let injectedCode = modifiedTemplateCode;
-  for (const { placeholder, alias } of Object.values(DATA_FETCHING_FUNCTIONS)) {
-    injectedCode = injectedCode.replace(placeholder, alias);
-  }
+    if (exportedIdentifiers.length > 0) {
+      outputFileContent += `export { ${exportedIdentifiers.join(', ')} } from "${
+        this.resourcePath
+      }?sentry-proxy-loader";`;
+    }
 
-  return `${modifiedUserCode}\n${injectedCode}`;
+    if (hasDefaultExport(ast)) {
+      outputFileContent += `
+        import { default as _sentry_default } from "${this.resourcePath}?sentry-proxy-loader";
+        import { withSentryGetInitialProps } from "@sentry/nextjs/build/esm/config/wrappers";
```

Comment:
Any reason not to import this straight from the SDK?

```suggestion
        import { withSentryGetInitialProps } from "@sentry/nextjs";
```

Reply:
Since `getInitialProps` can run on client and server I don't want to import the wrapper from the server sdk (or even the browser sdk for that matter) because I don't want to trigger side effects that are not isomorphic.

Good thing is, that when we mess up or move the wrappers file at some point, this will be a compile-time error.

Added a comment: 3ccbe1093e3d906968cae98ce1d60c2458f81d80

Reply:
Discussed: Importing from "@sentry/nextjs" is safe so let's do it

Reply:
Implemented in c0577da4ad77711f31e822dd5dce8490a2f59850